internal/events/handler.go

```diff
@@ -80,36 +78,16 @@ func (h *EventHandlerImpl) HandleEventBatch(ctx context.Context, batch EventBatc
 		return
 	}
 
-	err := h.updateNginx(ctx, conf)
-	if err != nil {
-		h.cfg.Logger.Error(err, "Failed to update NGINX configuration")
-	} else {
-		h.cfg.Logger.Info("NGINX configuration was successfully updated")
-	}
-
-	h.cfg.StatusUpdater.Update(ctx, statuses)
-}
-
-func (h *EventHandlerImpl) updateNginx(ctx context.Context, conf dataplane.Configuration) error {
-	// Write all secrets (nuke and pave).
-	// This will remove all secrets in the secrets directory before writing the requested secrets.
-	// FIXME(kate-osborn): We may want to rethink this approach in the future and write and remove secrets individually.
-	err := h.cfg.SecretMemoryManager.WriteAllRequestedSecrets()
-	if err != nil {
-		return err
-	}
-
-	cfg := h.cfg.Generator.Generate(conf)
+	h.changeCounter++
+	conf.Generation = h.changeCounter
 
-	// For now, we keep all http servers and upstreams in one config file.
-	// We might rethink that. For example, we can write each server to its file
-	// or group servers in some way.
-	err = h.cfg.NginxFileMgr.WriteHTTPConfig("http", cfg)
+	err := h.cfg.ConfigStorer.Store(conf)
 	if err != nil {
-		return err
+		h.cfg.Logger.Error(err, "error storing dataplane configuration")
+		// FIXME(kate-osborn): Update status to indicate that the gateway is not accepted or programmed.
```

Comment:
@pleshakov how do you feel about making this task a separate story to reduce the scope of this PR?